clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/VeniceKafkaInputReducer.java

```diff
@@ -43,10 +59,78 @@ public class VeniceKafkaInputReducer extends VeniceReducer {
 
   protected FilterChain<ChunkAssembler.ValueBytesAndSchemaId> veniceFilterChain;
 
+  private CompressorFactory compressorFactory;
+  private VeniceCompressor sourceVersionCompressor;
+  private VeniceCompressor destVersionCompressor;
+  private boolean passThrough = false;
+
   @Override
   protected void configureTask(VeniceProperties props, JobConf job) {
     super.configureTask(props, job);
     this.veniceFilterChain = initFilterChain(props);
+
+    compressorFactory = new CompressorFactory();
+    sourceVersionCompressor = getCompressor(
+        CompressionStrategy.valueOf(job.get(KAFKA_INPUT_SOURCE_COMPRESSION_STRATEGY)),
+        job.get(KAFKA_INPUT_BROKER_URL),
+        job.get(KAFKA_INPUT_TOPIC),
+        props);
+    destVersionCompressor = getCompressor(
```

Comment:
I think I'm a little confused about this.  If the source and destination versions are using zstd and this is a hybrid store, my intuition is that it's highly likely that we should never use passthrough (since the dictionary is potentially stale relative to data which came after the last full push).  I'm not sure I understand what gets retrieved on the destination version in a repush scenario?

Reply:
In VenicePushJob, there is an option to reuse the same dict: `kafka.input.compression.build.new.dict.enabled`, and it is enabled by default, and in some cases, we might want to reuse the dict in the repush, such as a repush to a batch-only store. 
Another case I can think of:
1. Regular push supports building a dict via MR job today (disabled by default).
2. This RB only adds support building a dict in Azkaban job thread, not via MR job.
3. So the dict built via MR job might be more efficient than the one built in repush.
In this case, we might want to reuse the original dict.

For real-time only stores, I think we should always build a new dict in a repush.

Reply:
Ok I get it now.  At this point we already populated the dictionary into the kafka topic, this is just making the call on weather or not we need to act on it.  Makes sense.